cmd/api/src/api/error.go

```diff
@@ -114,10 +118,15 @@ func BuildErrorResponse(httpStatus int, message string, request *http.Request) *
 // HandleDatabaseError writes an error (not found or other) depending on the database error encountered
 // Alternate: FormatDatabaseError()
 func HandleDatabaseError(request *http.Request, response http.ResponseWriter, err error) {
+	log.Infof("Handling database error: %v", err)
 	if errors.Is(err, database.ErrNotFound) {
 		WriteErrorResponse(request.Context(), BuildErrorResponse(http.StatusNotFound, ErrorResponseDetailsResourceNotFound, request), response)
 	} else if errors.Is(err, context.DeadlineExceeded) {
 		WriteErrorResponse(request.Context(), BuildErrorResponse(http.StatusInternalServerError, ErrorResponseRequestTimeout, request), response)
+	} else if strings.Contains(err.Error(), "duplicate key value violates unique constraint \"asset_groups_name_key\"") {
+		WriteErrorResponse(request.Context(), BuildErrorResponse(http.StatusConflict, ErrorResponseAGDuplicateName, request), response)
+	} else if strings.Contains(err.Error(), "duplicate key value violates unique constraint \"asset_groups_tag_key\"") {
+		WriteErrorResponse(request.Context(), BuildErrorResponse(http.StatusConflict, ErrorResponseAGDuplicateTag, request), response)
```

Comment:
This function is used in a number of areas outside of just asset groups, I think we either should generalize this further and / or find a home for these potentially higher up the stack. 

Futher, I think the `errors.Is()` is a must more elegant way to check these if it's possible to leverage that

Reply:
I've placed the checks in `CreateAssetGroup` in `bhce/cmd/api/src/api/v2/agi.go` to keep `HandleDatabaseError` more general.